unittests/files/keyusage.ts

```diff
@@ -1,3 +1,12 @@
+function assertType<T>(_x: T) {}
+
+const mockKey = {} as CryptoKey;
+assertType<Promise<ArrayBuffer | JsonWebKey>>(crypto.subtle.exportKey('' as KeyFormat, mockKey));
```

Comment:
IMO this is too loose as either `Promise<ArrayBuffer>` or `Promise<JsonWebKey>` will match the type. But not sure how to make it strict enough 🤔

Reply:
Maybe we can test whether it works for both `as Promise<ArrayBuffer>` and `as Promise<JsonWebKey>`, like:

```ts
crypto.subtle.exportKey('' as KeyFormat, mockKey) as Promise<ArrayBuffer>; // type includes ArrayBuffer
crypto.subtle.exportKey('' as KeyFormat, mockKey) as Promise<JsonWebKey>; // type also includes JsonWebKey
```

Reply:
> This is too loose as either Promise<ArrayBuffer> or Promise<JsonWebKey> will match the type

Isn't that exactly the point when `format` is not known to yield one or the other?

How about this?

```ts
assertType<Promise<ArrayBuffer | JsonWebKey>>(
  crypto.subtle
    .exportKey("" as KeyFormat, mockKey)
    .then((ambiguousKey) =>
      ambiguousKey instanceof ArrayBuffer
        ? (ambiguousKey satisfies ArrayBuffer)
        : (ambiguousKey satisfies JsonWebKey)
    )
);
```

Reply:
> Isn't that exactly the point when `format` is not known to yield one or the other?

Yes but that'll never fail when the return type regresses to either ArrayBuffer or JsonWebKey.

>How about this?

Sounds good!